lightning/src/ln/channelmanager.rs

```diff
@@ -240,6 +241,7 @@ impl MsgHandleErrInternal {
 					},
 				},
 			},
+			chan_id: Some(channel_id),
```

Comment:
We should not be setting this here - `send_err_msg_no_close` is used when we received a message from the wrong peer or otherwise don't have a channel to close, but need to send an error message. Thus, we shouldn't generate a ChannelClosed event.

Reply:
This should not generate a `ChannelClosed` as `shutdown_finish` is `None`. Though yes the `chan_id` is useless here, removed.